File: 윤석주/day6~7/큰 수 만들기.py
def solution(number, k):
    answer = []
    for i in range(len(number)):
        if k == 0:
            answer += number[i:]
            break
        
        while answer and k and answer[-1] < number[i]:
            answer.pop()
            k -= 1
        answer.append(number[i])
    
    while k:
        answer.pop()
        k-=1

    return ''.join(answer)

File: 박재현/큰수만들기.cpp
#include <string>
#include <vector>

using namespace std;

string solution(string number, int k)
{
    string answer = "";
    int size = number.size() - k;
    int endIndex = size;
    int index = 0;

    for (int i = 0; i < size; i++)
    {
        int startIndex;
        char maxN = '0';

        for (int j = index; j < number.size() - endIndex + 1; j++)
        { // 4 - 1 = 3
            if (maxN < number[j])
            {
                maxN = number[j];
                startIndex = j;
            }
        }
        index = startIndex + 1;
        endIndex--;
        answer += maxN;
    }

    return answer;
}